staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -309,6 +319,32 @@ func (w *watchCache) updateCache(event *watchCacheEvent) {
 	w.endIndex++
 }
 
+// Use dynamic size cache. If all cached events occur inside eventFreshDuration, increase cache capacity by 2x.
+// Decrease cache capacity by 2x when recent half events occur outside eventFreshDuration.
+func (w *watchCache) resizeCache(event *watchCacheEvent) {
+	if w.endIndex == w.startIndex+w.capacity && event.RecordTime.Sub(w.cache[w.startIndex%w.capacity].RecordTime) < eventFreshDuration {
+		if w.capacity*2 < maxEventCapacity {
+			w.makeNewCache(w.capacity * 2)
+			w.capacity *= 2
+		}
```

Comment:
instead of having "else if", let's add a return in the if

Reply:
addressed